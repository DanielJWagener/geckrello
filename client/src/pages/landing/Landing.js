import React from "react";

import setDefaultColors from "../../utilities/setDefaultColors";

class Landing extends React.Component {
  componentDidMount() {
    document.title = `Geckrello, a Trello Clone`;
    setDefaultColors();
  }

  render() {
    return <h1>Landing</h1>;
  }
}

export default Landing;
