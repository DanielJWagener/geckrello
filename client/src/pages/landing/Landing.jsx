import React from "react";

import setDefaultColors from "../../utilities/setDefaultColors";

import "./landing.styles.scss";

class Landing extends React.Component {
  componentDidMount() {
    document.title = `Geckrello, a Trello Clone`;
    setDefaultColors();
  }

  render() {
    return (
      <div className="landing">
        <div className="hero">
          <h1 className="hero__text">Your Life. Organized.</h1>
          <button className="hero__cta">Sign In</button>
        </div>
      </div>
    );
  }
}

export default Landing;
