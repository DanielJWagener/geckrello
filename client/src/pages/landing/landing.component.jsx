import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import setDefaultColors from "../../utilities/setDefaultColors";

import "./landing.styles.scss";

export class Landing extends React.Component {
  componentDidMount() {
    document.title = `Geckrello, a Trello Clone`;
    setDefaultColors();
  }

  renderCTA() {
    switch (this.props.auth) {
      case null:
        return <div className="hero__cta-invisible">placeholder</div>;
      case false:
        return (
          <a href="/auth/google" className="hero__cta">
            Sign In with Google
          </a>
        );
      default:
        return (
          <Link to="/dashboard" className="hero__cta">
            Go to Dashboard
          </Link>
        );
    }
  }

  render() {
    return (
      <div className="landing">
        <div className="hero">
          <h1 className="hero__text">
            <span className="hero__text--first">Your Life.</span>{" "}
            <span className="hero__text--second">Organized.</span>
          </h1>
          {this.renderCTA()}
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ auth }) => ({ auth });

export default connect(mapStateToProps)(Landing);
