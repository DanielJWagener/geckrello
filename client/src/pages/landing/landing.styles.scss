$color-primary: hsl(208, 79%, 51%);
$color-primary-dark: hsl(209, 100%, 35%);

@function hslac($color, $alpha) {
  @if (type-of($color) == "color") {
    @return hsla(hue($color), saturation($color), lightness($color), $alpha);
  } @else {
    @error "You didn't pass a color object";
  }
}

@keyframes moveIn {
  0% {
    opacity: 0;
    transform: translateY(-30rem);
  }

  80% {
    transform: translateY(-1rem);
  }

  100% {
    opacity: 1;
    transform: translateY(0);
  }
}

.landing {
  background-image: linear-gradient(
      to bottom,
      hslac($color-primary-dark, 1),
      hslac($color-primary, 0.6)
    ),
    url(/img/landing-hero-large.jpg);
  height: calc(var(--vh, 1vh) * 100 - var(--navbar-height));
  background-size: cover;
  display: flex;
  justify-content: center;

  .hero {
    align-self: center;
    display: flex;
    flex-direction: column;
    align-items: center;

    &__text {
      color: white;
      text-align: center;
      font-size: 7rem;
      font-family: inherit;
      font-weight: 200;
      margin-bottom: 4rem;

      &--first {
        animation-name: moveIn;
        animation-duration: 1.5s;
        animation-timing-function: ease-out;
      }

      &--second {
        animation-name: moveIn;
        animation-duration: 1.5s;
        animation-delay: 1s;
        animation-timing-function: ease-out;
        animation-fill-mode: backwards;
      }
    }

    &__cta {
      color: #e78632;
      text-align: center;
      text-decoration: none;
      border: 2px solid #e78632;
      border-radius: 0.5rem;
      background-color: #fff;
      width: max-content;
      padding: 1.5rem 2rem 1.5rem 2rem;
      font-size: 3rem;
      font-weight: 400;
      cursor: pointer;

      &--inivisible {
        opacity: 1;
      }

      &:hover {
        color: #e78632;
        border-color: #e78632;
        background-color: rgba(255, 255, 255, 0.8);
      }
    }
  }
}
