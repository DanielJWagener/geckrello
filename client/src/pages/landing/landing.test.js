import { shallow } from "enzyme";
import React from "react";

import { Landing as LandingPure } from "./landing.component";

it("renders Landing component", () => {
  expect(shallow(<LandingPure />).debug()).toMatchSnapshot();
});
