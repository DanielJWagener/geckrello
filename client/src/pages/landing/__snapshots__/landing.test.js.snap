// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Landing component 1`] = `
"<div className=\\"landing\\">
  <div className=\\"hero\\">
    <h1 className=\\"hero__text\\">
      <span className=\\"hero__text--first\\">
        Your Life.
      </span>
       
      <span className=\\"hero__text--second\\">
        Organized.
      </span>
    </h1>
    <Link to=\\"/dashboard\\" className=\\"hero__cta\\">
      Go to Dashboard
    </Link>
  </div>
</div>"
`;
